### PR TITLE
telemetry: add definitions for code scan changes

### PR DESCRIPTION
**Pull request by ctlai95**
## Problem

Continuation of #890 

## Solution

Add definitions for code scan changes

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/definitions/commonDefinitions.json [hunk 1]
```diff
@@ -4375,6 +4417,33 @@
                 }
             ]
         },
+        {
+            "name": "codewhisperer_codeScanIssueIgnore",
+            "description": "Called when a code scan issue is ignored",
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:4422]: the description is a good place to explain non-obvious fields. e.g. could mention : 
```suggestion
            "description": "User ignored a code scan issue. variant=all means the user ignored all issues. ",
```

File: telemetry/definitions/commonDefinitions.json [hunk 2]
```diff
@@ -6969,4 +7042,4 @@
             ]
         }
     ]
-}
+}
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:7045]: looks like your editor might be removing the final EOL. please check your editor/IDE settings

File: telemetry/definitions/commonDefinitions.json [hunk 3]
```diff
@@ -4340,15 +4362,35 @@
...
                 {
-                    "type": "reason",
+                    "type": "ruleId",
                     "required": false
+                }
+            ]
+        },
+        {
+            "name": "codewhisperer_codeScanIssueGenerateFix",
+            "description": "Called when a code scan issue fix is generated",
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:4372]:
```suggestion
            "description": "Generated a code scan issue. varant=... means ...",
```

File: telemetry/definitions/commonDefinitions.json (hunk 2 above)
Comment (ctlai95) [reply telemetry/definitions/commonDefinitions.json:7045]: Looks like it gets removed by running the validation script, but I can add it back manually

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**